libmamba/src/core/curl.cpp

```diff
@@ -4,12 +4,108 @@
 //
 // The full license is in the file LICENSE, distributed with this software.
 
+// TODO remove all these includes later?
 #include <spdlog/spdlog.h>
 
+#include "mamba/core/mamba_fs.hpp"  // for fs::exists
+#include "mamba/core/util.hpp"      // for hide_secrets
+
 #include "curl.hpp"
 
 namespace mamba
 {
+    namespace curl
+    {
+        void set_curl_handle(
```

Comment:
Totally agreed, I left it out of the `CURLHandle` class for now since it's used in `DownloadTarget::resource_exists()` and the given `handle` is not the one of the `CURLHandle` class.